### ISSUE TITLE
BUG: App incompatiblity with Camera cutout of new Essential PH-1

### ISSUE THREAD

**Issue by vivekkrish** [mozilla-mobile/focus-android#2047]
- Firefox Focus 4.0.1 (Build #13472157)

- [Essential PH-1](https://www.essential.com/)
	Android 8.0.0
	Security Patch Dec 5, 2017
	Build number OPM1.170911.213

- User agent: Mozilla/5.0 (Linux; Android 8.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/4.0.1 Chrome/63.0.3239.111 Mobile Safari/537.36

	Other HTTP headers:
	```	
	Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
	Accept-Encoding: gzip, deflate
	Accept-Language: en-US
	Host: duckduckgo.com
	User-Agent: Mozilla/5.0 (Linux; Android 8.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/4.0.1 Chrome/63.0.3239.111 Mobile Safari/537.36
	UPGRADE-INSECURE-REQUESTS: 1
	X-REQUESTED-WITH:
	```

- How to reproduce:
	1. Buy Essential PH-1
	2. Install Firefox Focus
	3. Profit?? 😝  
	
The app itself runs fine, but there are minor rendering issues of the Android status bar when the app is active (please see screenshots below). The height/padding of the bar is not being automatically adjusted to account for the unusually tall status bar on this phone (due to the in-display camera cutout).

Screenshot 1 (Homeview) | Screenshot 2 (Loaded page)
------------- | --------------
![Screenshot 1](https://i.imgur.com/gyCZM0Z.png) | ![Screenshot 2](https://i.imgur.com/1Ncj5T9.png)

Please do let me know if the above screenshots don't convey the issue properly (since you cannot see the camera cutout). If needed, I can take (and share) a photo of my actual phone running the app.

The dev team at Essential has asked owners of this phone to notify app developers directly when encountering this type of issue and make you aware of this thread on the [r/Essential](https://www.reddit.com/r/essential) subreddit (https://www.reddit.com/r/essential/comments/7amvxr/app_compatibility_with_camera_cutout_in_essential/).

In the OP of this thread, please refer to the section "For Developers", which explains the necessary steps to be taken to enable apps to dynamically set the status bar height.

I have already submitted this bug report via the Google Play Store feedback email (<android-marketplace-notices@mozilla.com>) over a month ago but received no response.

Hope you are able to review this request and fix this bug in an upcoming release.

Thank you very much for this wonderfully fast browser!

**Comment by pocmo** [comment]
Thank you for filing this issue. I always wondered how Focus would behave on this phone as they use a different status bar height.

> Buy Essential PH-1

I wish it would be available in Europe.. :)

**Comment by fercarcedo** [comment]
I don't have the PH-1, but I have been able to reproduce this bug on an Android P emulator thanks to the new feature of P of simulating a notch. 
I have also a fix for the problem ready, so I'm going to submit a PR.

**Comment by vivekkrish** [comment]
@pocmo, you are welcome! I've just heard recently that the Essential PH-1 will be released in the Netherlands, Belgium, and Luxembourg (https://www.reddit.com/r/essential/comments/80npri/essential_phone_is_coming_to_the_netherlands/), and likely other parts of Europe later this year!

@fercarcedo, I was just going to update this ticket after I read about the Android P Developer Preview release, saying that we can hope for native support soon! I see that you've already committed some code changes, awesome! I'm really looking forward to seeing your fix (#2264) incorporated into the code base.

**Comment by pocmo** [comment]
This landed and will be in the next release.

@vivekkrish If you want to test it. A new preview build should be available soon: https://bit.ly/focus-preview-builds

**Comment by vivekkrish** [comment]
This is amazing! Thank you so much @fercarcedo.
@pocmo, yes I can most definitely test out the preview build and provide feedback.

**Comment by vivekkrish** [comment]
@pocmo, just to confirm, is the version available at the URL below the one I should download and test?

https://index.taskcluster.net/v1/task/project.focus.android.preview-builds/artifacts/public/app-focus-webview-universal-beta-signed-aligned.apk

**Comment by pocmo** [comment]
@vivekkrish Yes, that's the correct version and it looks like there's a build with the changes available now!

**Comment by vivekkrish** [comment]
@pocmo and @fercarcedo, success! Please see the "after" screenshots below:

Screenshot 1 (Homeview) | Screenshot 2 (Loaded page)
------------- | --------------
![image](https://user-images.githubusercontent.com/711637/37223366-4008804a-239e-11e8-9ccb-c8c0983b994b.png) | ![image](https://user-images.githubusercontent.com/711637/37223375-482b11de-239e-11e8-9ac7-71c1deb2ee96.png)

Thank you so much once again, I'm very happy that this has been resolved and look forward to seeing this fix in the upcoming 4.1 release. 🎆

**Comment by pocmo** [comment]
@vivekkrish Awesome. Thank you for verifying!
